presto-main/src/main/java/io/prestosql/operator/scalar/MathFunctions.java

```diff
@@ -690,35 +690,50 @@ public static long round(@SqlType(StandardTypes.BIGINT) long num)
     @SqlType(StandardTypes.TINYINT)
     public static long roundTinyint(@SqlType(StandardTypes.TINYINT) long num, @SqlType(StandardTypes.INTEGER) long decimals)
     {
-        // TODO implement support for `decimals < 0`
-        return num;
+        return roundLong(num, decimals);
```

Comment:
You can go above tinyint max (or below tinyint min).
You can detect this with `com.google.common.primitives.SignedBytes#checkedCast`

See eg `io.prestosql.type.IntegerOperators#castToTinyint`.

Same for smallint and integer.